Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2722,50 +2731,33 @@ - (void)sendQualityAdjustedAttachmentForVideo:(NSURL *)movieURL
         presentFromViewController:self
                         canCancel:YES
                   backgroundBlock:^(ModalActivityIndicatorViewController *modalActivityIndicator) {
-                      AVAsset *video = [AVAsset assetWithURL:movieURL];
-                      AVAssetExportSession *exportSession =
-                          [AVAssetExportSession exportSessionWithAsset:video
-                                                            presetName:AVAssetExportPresetMediumQuality];
-                      exportSession.shouldOptimizeForNetworkUse = YES;
-                      exportSession.outputFileType = AVFileTypeMPEG4;
-                      NSURL *compressedVideoUrl = [[self videoTempFolder]
-                          URLByAppendingPathComponent:[[[NSUUID UUID] UUIDString]
-                                                          stringByAppendingPathExtension:@"mp4"]];
-                      exportSession.outputURL = compressedVideoUrl;
-                      [exportSession exportAsynchronouslyWithCompletionHandler:^{
-                          dispatch_async(dispatch_get_main_queue(), ^{
-                              OWSAssert([NSThread isMainThread]);
-
-                              if (modalActivityIndicator.wasCancelled) {
-                                  return;
+                      DataSource *dataSource = [DataSourcePath dataSourceWithURL:movieURL];
+                      dataSource.sourceFilename = filename;
+                      VideoCompressionResult *compressionResult =
+                          [SignalAttachment compressVideoAsMp4WithDataSource:dataSource
+                                                                     dataUTI:(NSString *)kUTTypeMPEG4];
+                      [compressionResult.attachmentPromise retainUntilComplete];
+
+                      compressionResult.attachmentPromise.then(^(SignalAttachment *attachment) {
```

Comment:
There's two `attachment` and two `dataSource` symbols of the same type in scope, right?  How about `original...`/`compressed...` or `src...`/`dst...` prefixes?

Reply:
Oh sorry, misleading diff.